ktor-io/common/src/io/ktor/utils/io/ByteReadChannelOperations.kt

```
@@ -450,6 +489,12 @@ public suspend fun ByteReadChannel.readUTF8LineTo(out: Appendable, max: Int = In
     }
 }
 
+private fun LineEndingMode.assertIncludes(other: LineEndingMode) {
+    if (other !in this) {
+        throw EOFException("Unexpected line ending $other, while expected $this")
+    }
+}
```

I don't think EOF is technically correct since it's malformed data but the input hasn't ended, maybe just an IOException would do.

Could there be scenarios here where you'd simply want to ignore the partial line ending?

---

> Could there be scenarios here where you'd simply want to ignore the partial line ending?

That is how Netty works. It just ignores everything until the LF line ending. It makes a server more tolerant to malformed requests. The question is, do we want to make it tolerant?

---

I think it's better to throw an exception as Jetty and Tomcat do it. Also, a user says that the behavior of Google Load Balancer (in Netty it's the same: parse input as one request with two chunks) is incorrect for them